packages/sqlite_async/lib/src/common/sqlite_database.dart

```diff
@@ -38,6 +38,21 @@ mixin SqliteDatabaseMixin implements SqliteConnection, SqliteQueries {
   ///
   /// Use this to access the database in background isolates.
   IsolateConnectionFactory isolateConnectionFactory();
+
+  /// TODO Improve on this definition by supporting a writeable context.
+  Future<void> exclusiveLock<T>(
+    Future<T> Function(SqliteReadContext ctx) callback,
+  ) {
+    return writeLock(callback);
+  }
+
+  /// Ensures that all connections are aware of the latest schema changes applied (if any).
+  /// Queries and watch calls can potentially use outdated schema information after a schema update.
+  Future<void> refreshSchema() {
+    return exclusiveLock((ctx) async {
+      return ctx.get("PRAGMA table_info('sqlite_master')");
+    });
+  }
```

Comment:
Do we actually need an exclusive lock across all connections?

I'm wondering if it's perhaps better to just run the pragma once on each connection, without worrying about it happening at the same time.

Reply:
I don't think I have a strong opinion on this, Steven helped me with the lock implementation.
But, doing the exclusive lock could be seen as a way to avoid potential race conditions downstream in Powersync when the schema is updated, and because `refreshSchema` is used infrequently, the performance impact of locking everything should be minimal right?

Reply:
Yeah, not a big performance impact - just feels like it's adding complexity and overhead that we don't actually need.

If we do keep the implementation, I'd consider making `exclusiveLock` an internal-only API if feasible, until we've properly fleshed out other use cases.